duplicate.cc

```diff
@@ -152,7 +152,7 @@ void DuplicateFilter::process()
 
   int ct = waypt_count();
 
-  auto* htable = (wpt_ptr*) xmalloc(ct * sizeof(wpt_ptr));
+  wpt_ptr htable[ct];
```

Comment:
I don't think this is legal in C++.  It was legal in C99.  g++ allows this as an extension:
https://gcc.gnu.org/onlinedocs/gcc/Variable-Length.html
but MSVC does not.

My preference would be with stick with the standard.

Reply:
QScopedArrayPointer works fairly nice here, although it can require some changes to how the pointer is used.